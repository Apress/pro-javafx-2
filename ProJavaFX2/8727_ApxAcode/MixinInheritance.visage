/*
 * @test
 * @run
 */

mixin class A {
  public var i:Integer;
  function f():Void {
    println("f() called: i = {i}.");
    i++;
  }
}

mixin class B {
  public var i: Integer;
  function f():Void {
    println("shadowed!");
  }
  function g() {
    println("g() called: i = {i}.");
    i++;
  }
}

class C extends A, B { }

var o = C { i: 1024 }
o.f();
o.g();
o.f();
