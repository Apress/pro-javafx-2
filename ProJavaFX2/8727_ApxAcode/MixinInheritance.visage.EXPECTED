f() called: i = 1024.
g() called: i = 1025.
f() called: i = 1026.
